src/main/java/org/broadinstitute/hellbender/utils/codecs/sampileup/SAMPileupFeature.java

```diff
@@ -0,0 +1,289 @@
+    }
+
+    public List<String> getFWDAlleles() {
+        return observedAlleles;
+    }
+
+    protected void setFWDAlleles(List<String> alleles) {
+        this.observedAlleles = alleles;
+    }
+
+    // ----------------------------------------------------------------------
+    //
+    // What kind of variant are we?
+    //
+    // ----------------------------------------------------------------------
+    public boolean isSNP() {
+        return varType == VariantType.SNP;
+    }
+
+    public boolean isInsertion() {
+        return varType == VariantType.INSERTION;
+    }
+
+    public boolean isDeletion() {
+        return varType == VariantType.DELETION;
+    }
+
+    public boolean isIndel() {
+        return isInsertion() || isDeletion() || varType == VariantType.INDEL;
+    }
+
+    public boolean isReference() {
+        return varType == VariantType.NONE;
+    }
+
+    protected void setVariantType(VariantType variantType) {
+        this.varType = variantType;
+    }
+
+    public boolean isHom() {
+        // implementation-dependent: here we use the fact that for ref and snps we actually use fixed static strings to remember the genotype
+        if (!isIndel()) {
+			return (observedAlleles.get(0).equals(observedAlleles.get(1)));
+		}
+        return (isInsertion() || isDeletion()) && observedAlleles.get(0).equals(observedAlleles.get(1));
+    }
+
+    public boolean isHet() {
+        // implementation-dependent: here we use the fact that for ref and snps we actually use fixed static strings to remember the genotype
+        if (!isIndel()) {
+			return (!(observedAlleles.get(0).equals(observedAlleles.get(1))));
+		}
+        return isIndel() || (!observedAlleles.get(0).equals(observedAlleles.get(1)));
+    }
+
+    public double getVariantConfidence() {
+        return variantScore;
+    }
+
+    protected void setVariantConfidence(double variantScore) {
+        this.variantScore = variantScore;
+    }
+
+    public boolean isBiallelic() {
+        return nNonref < 2;
+    }
+
+    protected void setNumNonRef(int nNonref) {
+        this.nNonref = nNonref;
+    }
+
+    public double getConsensusConfidence() {
+        return consensusScore;
+    }
+
+    protected void setConsensusConfidence(double consensusScore) {
+        this.consensusScore = consensusScore;
+    }
+
+    public int length() {
+        return eventLength;
+    }
+
+    protected void setLength(int eventLength) {
+        this.eventLength = eventLength;
+    }
+
+    public boolean isIndelGenotype() {
+        return refBaseChar == '*';
+    }
+
+    public boolean isPointGenotype() {
+        return !isIndelGenotype();
+    }
+
+    /**
+     * Implements method required by GenotypeList interface. If this object represents an indel genotype, then it
+     * returns itself through this method. If this object is a point genotype, this method returns null.
+     *
+     * @return
+     */
+    public SAMPileupFeature getIndelGenotype() {
```

Comment:
a lot of these are for some outdated GenotypeList interface that aren't needed anymore and don't really do anything